jsoncpp-namespaced-targets.cmake

```diff
@@ -1,7 +1,7 @@
-if (TARGET jsoncpp_static)
+if (TARGET jsoncpp_static AND NOT TARGET JsonCpp::JsonCpp)
```

Comment:
ok I had to look this up. `if (TARGET x)` tests for the existence of a definition of a target `x`.

https://cmake.org/cmake/help/latest/command/if.html#target

So if I understand correctly, this is saying that before we call `add_library` to create a `JsonCpp::JsonCpp` target, we are going to check whether one already exists.

I'm a cmake newbie, but make this is better as its own if branch above the others.
Then we are only changing 1 line and we aren' repeating ourselves as much.

```cmake
if (TARGET JsonCpp::JsonCpp)
elseif (TARGET jsoncpp_static)
    ...
elseif (TARGET jsoncpp_lib AND NOT TARGET JsonCpp::JsonCpp)
    ...
endif ()
```

If that doesn't work, that's fine too.

Reply:
@rjoomen Would the requested change work?

Reply:
I suppose that would work and is slightly cleaner. I modified the PR.